src/main/scala/sangria/relay/Connection.scala

```diff
@@ -1,18 +1,17 @@
 package sangria.relay
 
-import sangria.execution.{ExecutionError, UserFacingError}
-
-import language.higherKinds
+import sangria.execution.UserFacingError
 import sangria.relay.util.Base64
 import sangria.schema._
 
-import scala.concurrent.{ExecutionContext, Future}
+import scala.annotation.implicitNotFound
+import scala.concurrent.{ ExecutionContext, Future }
+import scala.language.higherKinds
 import scala.reflect.ClassTag
 import scala.util.Try
-import scala.annotation.implicitNotFound
 
-trait Connection[T] {
-  def pageInfo: PageInfo
+trait Connection[+T, +P <: PageInfo] {
```

Comment:
I'm not sure you need to introduce `P` type parameter on `Connection`. I think we can stick to `PageInfo` here. `Connection.definition` relies on `ConnectionLike` type-class, and I think this is the only place that needs a change, isn't it?

Reply:
I did'nt realize `ConnectionLike` was a type-class for this usecase. Thanks!